src/audio/streaminfo.cpp

```diff
@@ -0,0 +1,27 @@
+#include "audio/streaminfo.h"
+
+namespace mixxx {
+
+namespace audio {
+
+bool operator==(
+        const StreamInfo& lhs,
+        const StreamInfo& rhs) {
+    return lhs.getSignalInfo() == rhs.getSignalInfo() &&
+            lhs.getBitrate() == rhs.getBitrate() &&
+            lhs.getDuration() == rhs.getDuration();
+}
+
+QDebug
+operator<<(QDebug dbg, const StreamInfo& arg) {
```

Comment:
Thanks for taking the time to make all these QDebug operator overloads. I'm sure it will come in handy.

Reply:
Lessons learned from Rust. There support for debug info only requires to add a _derive_ attribute and everything else just works.

Reply:
at least we have `std::optional` in C++ now